instrumentation/httpasyncclient/src/test/java/brave/httpasyncclient/ITTracingHttpAsyncClientBuilder.java

```diff
@@ -92,4 +98,54 @@ protected void post(CloseableHttpAsyncClient client, String pathIncludingQuery,
     
     takeSpan();
   }
+
+  @Test public void currentSpanIsVisibleInCallbackThread() throws Exception {
```

Comment:
I would go a bit farther, because you are not only setting a current span, but specifically the invocation context. Ex if you change spanIsVisible to something that catches the current tracecontext, you can compare it against the one in line 124